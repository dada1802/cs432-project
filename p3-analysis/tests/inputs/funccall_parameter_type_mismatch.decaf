def int add2(int a)
{
    return a + 2;
}

def int main()
{
    int a;
    a = add2(false);

    return 0;
}