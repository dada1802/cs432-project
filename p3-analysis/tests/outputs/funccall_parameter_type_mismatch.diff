--- outputs/funccall_parameter_type_mismatch.txt	2023-10-26 02:38:54.291612054 -0400
+++ expected/funccall_parameter_type_mismatch.txt	2023-10-25 22:04:13.847777986 -0400
@@ -1,22 +1 @@
-Program [line 1]
-SYM TABLE:
- print_int : (int) -> void
- print_bool : (bool) -> void
- print_str : (str) -> void
- add2 : (int) -> int
- main : () -> int
-
-  FuncDecl name="add2" return_type=int parameters={a:int} [line 1]
-  SYM TABLE:
-   a : int
-
-    Block [line 2]
-    SYM TABLE:
-
-  FuncDecl name="main" return_type=int parameters={} [line 6]
-  SYM TABLE:
-
-    Block [line 7]
-    SYM TABLE:
-     a : int
-
+Type mismatch in parameter 0 of call to 'add2': expected int but found bool on line 9
