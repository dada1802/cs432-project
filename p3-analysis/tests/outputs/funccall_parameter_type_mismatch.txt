Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 add2 : (int) -> int
 main : () -> int

  FuncDecl name="add2" return_type=int parameters={a:int} [line 1]
  SYM TABLE:
   a : int

    Block [line 2]
    SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 6]
  SYM TABLE:

    Block [line 7]
    SYM TABLE:
     a : int

