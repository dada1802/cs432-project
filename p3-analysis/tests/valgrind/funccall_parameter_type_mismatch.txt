==2611514== Memcheck, a memory error detector
==2611514== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2611514== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==2611514== Command: ../decaf inputs/funccall_parameter_type_mismatch.decaf
==2611514== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 add2 : (int) -> int
 main : () -> int

  FuncDecl name="add2" return_type=int parameters={a:int} [line 1]
  SYM TABLE:
   a : int

    Block [line 2]
    SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 6]
  SYM TABLE:

    Block [line 7]
    SYM TABLE:
     a : int

==2611514== 
==2611514== HEAP SUMMARY:
==2611514==     in use at exit: 0 bytes in 0 blocks
==2611514==   total heap usage: 1,916 allocs, 1,916 frees, 233,016 bytes allocated
==2611514== 
==2611514== All heap blocks were freed -- no leaks are possible
==2611514== 
==2611514== For lists of detected and suppressed errors, rerun with: -s
==2611514== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
