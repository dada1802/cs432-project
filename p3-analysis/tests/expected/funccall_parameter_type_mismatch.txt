Type mismatch in parameter 0 of call to 'add2': expected int but found bool on line 9
